fuel-block-producer/src/config.rs

```diff
@@ -1,2 +1,25 @@
-#[derive(Default, Debug, Clone)]
-pub struct Config {}
+use fuel_core_interfaces::common::{
+    fuel_tx::ConsensusParameters,
+    fuel_types::{
+        Address,
+        Word,
+    },
+};
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    /// The primary identity associated with the validator (i.e. staking key)
+    pub validator_id: Address,
+    pub max_gas_per_block: Word,
```

Comment:
Maybe it should be part of the consensus?

Reply:
Ah good point, this totally seems like something that should be part of consensus params.

I'll make a PR to fuel-tx for this..

Reply:
As we discussed on slack, we should bundle consensus parameters with the appropriate domain and then aggregate all the various domain specific settings in fuel-core. This will prevent fields being added to fuel-tx that are unused by fuel-tx or the VM, and avoid causing breaking changes in fuel tx when settings unrelated to tx validation are updated.